src/lib/components/UrlChooser/index.js

```diff
@@ -0,0 +1,185 @@
+
+    store.subscribe(this.onStateChanged.bind(this));
+    this.onStateChanged();
+  }
+
+  render() {
+    return html`
+      <div class="report_header_enterurl">
+        <div class="lh-enterurl lh-enterurl--selected">${this.href}</div>
+        <div class="lh-enterurl lh-enterurl--switch">
+          <input
+            type="url"
+            class="lh-input"
+            .value="${this.href}"
+            placeholder="Enter a web page URL"
+            pattern="https?://.*"
+            minlength="7"
+            required
+            @keyup="${this.onUrlKeyup}"
+          />
+          <button
+            class="lh-enterurl__close"
+            aria-label="Remove URL"
+            @click=${this.clearInput}
+          ></button>
+        </div>
+        <div class="lh-controls">
+          <button
+            ?disabled=${this.active}
+            class="w-button w-button--secondary"
+            @click=${this.switchUrl}
+          >
+            Switch URL
+          </button>
+          <button
+            ?disabled=${this.active}
+            class="w-button w-button--primary"
+            id="run-lh-button"
+            @click=${this.runAudit}
+          >
+            Run Audit
+          </button>
+        </div>
+      </div>
+    `;
+  }
+
+  updated() {
+    this.urlInput = this.renderRoot.querySelector('input[type="url"]');
+    this.runLighthouseButton = this.renderRoot.querySelector("#run-lh-button");
+  }
+
+  onStateChanged() {
+    const state = store.getState();
+    const urlInputValue = this.urlInput ? this.urlInput.value : null;
+
+    if (this.active) {
+      // do nothing, the URL is being run through Lighthouse
+    } else if (
+      this.href == null &&
+      this.switching &&
+      state.userUrl &&
+      !urlInputValue
+    ) {
+      // if the user has just signed in, the element was in an initial state,
+      // AND the user hasn"t typed anything, reset element with URL
+      this.href = state.userUrl;
+      this.switching = false;
+    } else if (state.userUrl == null && !this.switching) {
+      // if the user has signed out, clear the href and enter switching mode
+      this.href = null;
+      this.switching = true;
+    } else if (state.userUrl && !this.switching) {
+      // in all other cases, only update the URL if ther user isn"t switching
+      this.href = state.userUrl;
+    }
+  }
+
+  runAudit(ev) {
+    if (this.switching) {
+      if (!this.fixUpAndValidateUrl()) {
+        return;
+      }
+
+      // write URL to firestore and local
+      const url = this.urlInput.value;
+      store.setState({userUrl: url});
+      const ref = userRef();
+      if (ref) {
+        const p = ref.set(
+          {
+            userUrl: url,
+            userUrlUpdate: firebase.firestore.FieldValue.serverTimestamp(),
+          },
+          {merge: true},
+        );
+        p.catch((err) => {
+          // TODO: Firestore errors are problematic but we can still run with the new URL.
```

Comment:
Is there a TODO here or is this more of an informational note about how to handle the failure?

Reply:
A bit of both. My point is we don't really care if this fails, because we're still going to run Lighthouse, but the user won't see the same URL on their next visit.

Reply:
Maybe the comment should say "Note:"? I guess since we don't have any plans to change this behavior